src/dodal/devices/i19/blueapi_device.py

```diff
@@ -0,0 +1,102 @@
+        self.headers = HEADERS
+        super().__init__(name)
+
+    @AsyncStatus.wrap
+    async def set(self, value: D):
+        """ On set send a POST request to the optics blueapi with the name and \
+        parameters, gets the generated task_id and then sends a PUT request that runs \
+        the plan.
+
+        Args:
+            value (dict): The value passed here should be the parameters for the POST \
+                request, taking the form:
+                {
+                    "name": "plan_name",
+                    "params": {
+                        "experiment_hutch": f"{hutch_name}",
+                        "access_device": "access_control",
+                        "other_params": "...",
+                        ...
+                    }
+                }
+        """
+        # Value here vould be request params dictionary.
+        request_params = json.dumps(value)
+
+        async with ClientSession(base_url=self.url, raise_for_status=True) as session:
+            # First submit the plan to the worker
+            async with session.post(
+                "/tasks", data=request_params, headers=HEADERS
+            ) as response:
+                LOGGER.info(
+                    f"Task submitted to the worker, response status: {response.status}"
+                )
+
+                try:
+                    data = await response.json()
+                    task_id = data["task_id"]
+                except Exception as e:
+                    LOGGER.error(
+                        f"Failed to get task_id from {self.url}/tasks POST. ({e})"
+                    )
+                    raise
+            # Then set the task as active and run asap
+            async with session.put(
+                "/worker/task", data=json.dumps({"task_id": task_id}), headers=HEADERS
+            ) as response:
+                if not response.ok:
+                    LOGGER.error(
+                        f"""Session PUT responded with {response.status}: {response.reason}.
+                        Unable to run plan {value["name"]}."""  # type: ignore
+                    )
+                    return
+                LOGGER.info(f"Running plan: {value['name']}, task_id: {task_id}")  # type: ignore
+
+            # Poll server at 2Hz until plan complete or errored
+            interval = 0.5
+            plan_complete = False
+
+            while not plan_complete:
+                async with session.get(f"/tasks/{task_id}") as res:
+                    plan_result = await res.json()
+                    plan_complete = plan_result["is_complete"]
+                    errors = plan_result["errors"]
+                    if len(errors) > 0:
+                        message = "\n".join(errors)
+                        LOGGER.error(f"Plan {value['name']} failed: {message}")  # type:ignore
+                        raise RuntimeError(f"Plan failed with error: {message}")
+                    time.sleep(interval)  # noqa
```

Comment:
Must: This is an async function so asyncio sleep is fine

```suggestion
                    await asyncio.sleep(interval)
```

Also could: Move outside of the context block, otherwise you're keeping the HTTP connection alive while you're not doing anything (if my read of the [aiohttp client docs](https://docs.aiohttp.org/en/stable/client_advanced.html) is correct).

```suggestion
                await asyncio.sleep(interval)
```